### ISSUE TITLE
[weather_data_assimilation] cannot replicate training results

### ISSUE THREAD

**Issue by citronella3alain** [as2c/NeuralDMD#1]
tl;dr: weather data assimilation inference works with provided weights, but training yields bad results

While running `test_model.py` with the provided weights yields expected results, I can't seem to get training (`train_model.py`) to work well be it atop the provided weights or from scratch; are there any parameters to update, e.g., learning rate, seed, etc., in order to get it to work?

This is the reconstruction gif that I'm getting:

![Image](https://github.com/user-attachments/assets/4c548d09-565b-48c1-a26a-fd183b511438)
timestep 0:

<img width="1580" height="869" alt="Image" src="https://github.com/user-attachments/assets/2707b2e9-fa3f-414d-af71-bbd12e751873" />

timestep 160:

<img width="1580" height="869" alt="Image" src="https://github.com/user-attachments/assets/062318db-2f2b-40d8-9719-7218c893ebb9" />

Here's the loss plot I'm getting:

<img width="1000" height="600" alt="Image" src="https://github.com/user-attachments/assets/4eea173c-c11b-44b8-b537-6433042bea9c" />

I think it's interesting that the loss plot looks "normal" and that the loss starts out so high for the existing model weights.

Let me know what other info I can provide / what else I should try. Thanks!

**Comment by danperazzo** [comment]
Hello!!

Do you have any updates on this? I was also trying to replicate the results of the paper and also had the same problem.

**Comment by glitch401** [comment]
@as2c has there been any resolution to this issue?
